pkg/operation/botanist/monitoring.go

```diff
@@ -448,6 +434,38 @@ func (b *Botanist) DeleteSeedMonitoring(ctx context.Context) error {
 				Name:      "kube-state-metrics-seed-vpa",
 			},
 		},
+	} {
+		if err := b.K8sSeedClient.Client().Delete(ctx, obj); client.IgnoreNotFound(err) != nil && !meta.IsNoMatchError(err) {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// DeleteSeedMonitoring will delete the monitoring stack from the Seed cluster to avoid phantom alerts
+// during the deletion process. More precisely, the Alertmanager and Prometheus StatefulSets will be
+// deleted.
+func (b *Botanist) DeleteSeedMonitoring(ctx context.Context) error {
```

Comment:
Shouldn't this function call `DeleteKubeStateMetricsSeed`?

Reply:
`DeleteKubeStateMetricsSeed` is called in the `DeploySeedMonitoring` function to remove any existing `kube-state-metrics-seed` deployments running in the control plane. Why should `DeleteSeedMonitoring` call this function?

Reply:
If an existing shoot already has a deletion timestamp and the new Gardener version with this change is deployed then `DeploySeedMonitoring` won't be called, but only `DeleteSeedMonitoring`. In this case the `kube-state-metrics-seed` won't be deleted, or?

Reply:
Good catch, I will call the function in `DeleteSeedMonitoring`